### PR TITLE
fix(eigen_bundler): add tests and fix edge case during bundling

### PR DESCRIPTION
**Pull request by rymnc**

### CODE REVIEW

File: packages/services/src/block_bundler/eigen_bundler.rs [hunk 1]
```diff
@@ -149,10 +165,20 @@ impl EigenBundler {
                         current_blocks.len(),
                         original_count
                     );
-                    return (current_blocks, Some(bundle), Some(fragments));
+                    return Ok((current_blocks, Some(bundle), Some(fragments)));
                 }
             }
 
+            if current_blocks.len() == 1 {
```
Comment (rymnc) [inline packages/services/src/block_bundler/eigen_bundler.rs:172]: without this check, it would be stuck in an infinite loop

**Review by Dentosal — APPROVED** [review]
